docs/ruleset_functions/get_zone_target_baseline_system.md

```diff
@@ -0,0 +1,131 @@
+- otherwise if the building is 5 floors or fewer and between 25,000 and 150,000 sf: `elif( num_floors < 6 && area >= 25000 && area <= 150000 ):`
+	- then the system type is also SYS-5: `G3_1_1d_expected_system_type = SYS_5`
+- all other cases: `else:`
+	- the system type is SYS-7: `G3_1_1d_expected_system_type = SYS_7`
+- loop through the zones_and_systems to see if any of the zones meets G3.1.1.d: `for zone in zones_and_systems:`
+	- use the function does_zone_meet_G3_1_1d to determine if this zone meets this requirement: `if does_zone_meet_G3_1_1d(B_RMI,zone,zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] == YES):`
+		- change the system origin string: `zones_and_systems[zone]["SYSTEM_ORIGIN"] = "G3_1_1d"` 
+		- set the system to SYS_5 or SYS_7: `zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] = G3_1_1d_expected_system_type`
+
+- G3.1.1e "Thermal zones designed with heating-only systems in the proposed design serving storage rooms, stairwells, vestibules, electrical/mechanical rooms, and restrooms not exhausting or transferring air from mechanically cooled thermal zones in the proposed design shall use system type 9 or 10 in the baseline building design."
+- loop through the zones_and_systems to see if any of the zones meets this requirement: `for zone in zones_and_systems:`
+	- use the function does_zone_meet_G3_1_1e to determine whether to change the system type (we are using != NO instead of YES, because the function can return MAYBE.  Here we will assign any "maybe's to the new system type.  The rule itself will deal with the uncertainty of vestibules using the function is_zone_a_vestibule): `if does_zone_meet_G3_1_1e(P-RMI,B-RMI,zone) != NO:`
+		- - change the system origin string: `zones_and_systems[zone]["SYSTEM_ORIGIN"] = "G3_1_1e"`
+		- choose system 9 or 10 based on climate zone: `if is_CZ_0_to_3a():`
+			- set the system to SYS_10: `zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] = SYS_10`
+		- else (the climate zone is_CZ_3b_3c_and_4_to_8): `else:`
+			- set the system to SYS_9: `zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] = SYS_9`
+
+
+- G3.1.1f "If the baseline HVAC system type is 9 or 10, use additional system types for all HVAC zones that are mechanically cooled in the proposed design."
+- loop through the zones and systems, check only the zones that are SYS-9 or 10: `for zone in zones_and_systems:`
+	- check if it is SYS-9 or SYS-10: `if zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] in [SYS_9, SYS_10]:`
+		- use the function does_zone_meet_G3_1_1f to see if the zone meets the requirements of G3_1_1f: `if does_zone_meet_G3_1_1f(P-RMI,B-RMI,zone.id) == YES:`
+			- change the system origin string: `zones_and_systems[zone]["SYSTEM_ORIGIN"] = "G3_1_1f"`
+			- now determine the baseline system type using the zone HVAC building area type, the total building area and the total building number of floors.  First by getting the bat by looping through the building area types: `for bat in list_building_area_types_and_zones:`
```

Comment:
The example that I have previously provided (and that is shown in "Conversations" tab but not on this tab) is a warehouse project with some office zones that have cooling. Consider the two scenarios below. The logic works for #1 but not for #2:
1. If the project is treated as a mixed use building, have separate BAT specified for warehouse zones versus office zones, and office zones do not qualify as non-predominant condition (<20K SF), the baseline HVAC system type would be properly updated.
2. If the project is not treated as a mixed use building and has "warehouse" BAT assigned to all zones, your logic with will result in mapping office zones back to Sysem 9/10 instead of 3-8. 

I would updat the logic to check if the mapping again points to 9/10, and if so establish BAT based on zone types. You can do that if you isolate the portion of get_HVAC_building_area_types_and_zones that starts with "else look at each zone, and then each space and determine the building_segment_HVAC_BAT using the largest space.lighting_space_type" into a separate function and call it here (and modify get_HVAC_building_area_types_and_zones to call this function also).

Reply:
@mkarpman - I started implementing this change, but it is not quite that simple to break out the mapping zone-by zone.  if there is no building segment HVAC_BAT, get_HVAC_building_area_types_and_zones looks at each space lighting type and area, then assigns the building segment HVAC_BAT based on the largest HVAC_BAT as calculated based on spaces.  Let's discuss on the call.